ui/app/services/capabilities.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Service, { service } from '@ember/service';
+
+import type StoreService from 'vault/services/store';
+
+export default class CapabilitiesService extends Service {
+  @service declare readonly store: StoreService;
+
+  request = (apiPath: string) => {
+    return this.store.findRecord('capabilities', apiPath);
+  };
+
+  async fetchAll(apiPath: string) {
+    try {
+      return await this.request(apiPath);
+    } catch (e) {
+      return e;
+    }
+  }
+
+  async fetchSpecific(apiPath: string, capability: string) {
+    try {
+      const capabilities = await this.request(apiPath);
+      return capabilities[capability];
+    } catch (e) {
+      return e;
+    }
+  }
```

Comment:
my thought was having this be an internal method, and the service would typically call `canRead` or `canUpdate`

Reply:
Yeah! I was planning to add the underscore if folks agreed with this approach. Thanks for the feedback!